### ISSUE TITLE
Instance-level silence under some circumstances

### ISSUE THREAD

**Issue by Wonderfall** [tootsuite/mastodon#4721]
That was an issue I had in mind since the beginning but now I have to face it almost weekly if not daily. Following this thread I made : https://social.targaryen.house/@wonderfall/5363002 (we also tried in DMs)

I think that many persons are not aware that silence does not only impact the federated timeline. Indeed silencing also prevents the silenced user from appearing in notifications (if that user is not followed). That is logical, since this feature was also designed to prevent global harassment from a user.

Since many admins are using a lot this feature at an instance level (rest assured I'm not here to discuss if that's a good thing or not), it's up to each admin to be transparent or not (like keeping updated a list of silenced/blocked instances). But I don't think it's enough, and I think Mastodon itself can improve that and make things better.

*Instance X silenced instance Y.
Alice is a regular user from instance X. Alice may not be aware, for some reason, that instance Y was added to the silence list of instance X. Alice wants to ask something to Bob, who is on instance Y, because (let's say) Bob is a maintainer of an app Alice is using. Alice sends a toot, and Bob gets notified. Bob answers to Alice, but Alice will likely never see Bob's answer, because instance X silenced instance Y. Alice could follow Bob, yeah, but why would she do that? 
(Then, Alice will think Bob is an asshole because Alice couldn't see the reply. End of story.)*

Okay, that was just a particular example, but I can make other stories like that, where Alice would like to get notified from Bob, but there is no other way than following Bob (and again, why Alice would do that if Alice is not aware of the instance-level silence).

My proposition is simple. When mentioning a silenced user, there should be at least a warning that a notification of a possible answer from that silenced user won't be received, unless you decide to follow that user. A parallel proposition would be the possibility to unsilence a user or an instance manually, in order to give the final choice to the user (isn't that the purpose of Mastodon, to give social networking back to everyone?). But that also implies that Mastodon would make the process far more transparent than it is actually (i.e. make systematically the silence/blocklist public and available).

But that's just my propositions. You can address this issue differently, or not at all, but I think you should take a look at it, I'm opening this issue so we can discuss that.

**Comment by nightpool** [comment]
i don't understand the title/scope of this issue. Are you suggesting a feature? discussing a problem? It would be helpful if you edited the title of this issue to better summarize it's contents and the resolution you're hoping to see.

**Comment by Wonderfall** [comment]
I apologise, I thought the title would be clear enough. This issue concerns "instance-level silence under some circumstances" and I can't describe them all here. It can be both a feature suggestion and a problem. At the beginning the title was an explicit feature request, but then I came with the conclusion that there are many different approaches to this problem. So I didn't want to impose a single feature, I did want to open a discussion about that, suggest some solutions, that's all.

I have no better idea for a name at the moment, but I'm thinking about it. :wink:
